storage/innobase/dict/dict0dict.cc

```diff
@@ -7450,27 +7450,83 @@ dberr_t dict_get_dictionary_info_by_id(ulint dict_id, char **name,
   return err;
 }
 
-bool dict_detect_encryption(bool is_upgrade, space_id_t mysql_plugin_space) {
-  bool encrypt_mysql = false;
-  if (is_upgrade) {
-    if (mysql_plugin_space == SYSTEM_TABLE_SPACE) {
-      return (srv_sys_space.is_encrypted());
-    }
+static bool dict_should_mysql_ibd_be_created_encrypted_due_to_upgrade(
```

Comment:
Please keep the function names precise. Exact details of function can be added as function comment.

Reply:
I disagree. Comments then need to change when function changes. I took this from clean code and I will stand by it. Self explanatory names are much better - imho.

Reply:
I disagree. Function names should describe what the function is doing. Comments are additional maintenance burden and thus should be avoided if possible. Thus the code should be self explanatory. I have changed the names to be more generic and describe what the function is doing instead of describing what it is used for.

Reply:
Ok, I disagree too. See my comment in the place where you use "is_on_or_force()"

Reply:
I think it was agreed that we are not code reviewing individual codying styles. I opt for longer names describing what a function is doing. You can disagree, but you cannot hold PR because of that.

Reply:
Anyways, you can keep this as you wish.

Am not holding this PR for this reason. There are other comments, concerns raised. If those are solved, it can proceed further.